pos_fixed_discount/static/src/js/discount.js

```diff
@@ -0,0 +1,89 @@
+odoo.define("pos_fixed_discount.FixedDiscountButton", function (require) {
+    "use strict";
+
+    const PosComponent = require("point_of_sale.PosComponent");
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const {useListener} = require("web.custom_hooks");
+    const Registries = require("point_of_sale.Registries");
+
+    class FixedDiscountButton extends PosComponent {
+        constructor() {
+            super(...arguments);
+            useListener("click", this.onClick);
+        }
+        async onClick() {
+            var self = this;
+            const {confirmed, payload} = await this.showPopup("NumberPopup", {
+                title: this.env._t("Discount Amount"),
+                startingValue: 0,
+            });
+            if (confirmed) {
+                var val = Math.round(Math.max(0, Math.min(100, parseFloat(payload))));
+                self.apply_discount(val);
```

Comment:
```suggestion
                this.apply_discount(val);
```

You can do this and drop the `var self = this;` line at the top